crypto/src/utils.rs

```diff
@@ -0,0 +1,205 @@
+// limitations under the License.
+
+#![allow(dead_code)] // TODO Remove after complete ECR-2518 and  ECR-2519
+use super::{gen_keypair, gen_keypair_from_seed, PublicKey, SecretKey, Seed, SEED_LENGTH};
+use hex_buffer_serde::Hex;
+use pwbox::{sodium::Sodium, ErasedPwBox, Eraser, Suite};
+use rand::thread_rng;
+use std::borrow::Cow;
+#[cfg(unix)]
+use std::os::unix::fs::{MetadataExt, OpenOptionsExt};
+use std::{
+    fs::{File, OpenOptions},
+    io::{Error, ErrorKind, Read, Write},
+    path::Path,
+};
+use toml;
+
+/// Creates a TOML file that contains encrypted `SecretKey` and returns `PublicKey` for the secret key.
+pub fn create_keys_file<P: AsRef<Path>>(path: P, pass_phrase: &[u8]) -> Result<PublicKey, Error> {
+    let (pk, sk) = gen_keypair();
+    let keys = EncryptedKeys::encrypt(pk, &sk, pass_phrase)?;
+    let file_content =
+        toml::to_string_pretty(&keys).map_err(|e| Error::new(ErrorKind::Other, e))?;
+    let mut open_options = OpenOptions::new();
+    open_options.create(true).write(true);
+    #[cfg(unix)]
+    open_options.mode(0o600);
+    let mut file = open_options.open(path.as_ref())?;
+    file.write_all(file_content.as_bytes())?;
+
+    Ok(pk)
+}
+
+/// Reads `PublicKey` and `SecretKey` from encrypted file located by path and returns its.
+pub fn read_keys_from_file<P: AsRef<Path>>(
+    path: P,
+    pass_phrase: &[u8],
+) -> Result<(PublicKey, SecretKey), Error> {
+    let mut key_file = File::open(path)?;
+
+    if cfg!(unix) {
+        let file_info = key_file.metadata()?;
+        if (file_info.mode() & 0o600) != 0o600 {
+            return Err(Error::new(ErrorKind::Other, "Wrong file's mode"));
+        }
+    }
+
+    let mut file_content = vec![];
+    key_file.read_to_end(&mut file_content)?;
+    let keys: EncryptedKeys =
+        toml::from_slice(file_content.as_slice()).map_err(|e| Error::new(ErrorKind::Other, e))?;
+    keys.decrypt(pass_phrase)
+}
+
+struct PublicKeyHex;
+
+impl Hex<PublicKey> for PublicKeyHex {
+    fn create_bytes(value: &PublicKey) -> Cow<[u8]> {
+        Cow::Borrowed(&*value.as_ref())
+    }
+
+    fn from_bytes(bytes: &[u8]) -> Result<PublicKey, String> {
+        PublicKey::from_slice(bytes)
+            .ok_or_else(|| "Couldn't create PublicKey from slice".to_owned())
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+struct EncryptedKeys {
+    #[serde(with = "PublicKeyHex")]
+    public_key: PublicKey,
+    secret_key: ErasedPwBox,
+}
+
+impl EncryptedKeys {
+    fn encrypt(
+        public_key: PublicKey,
+        secret_key: &SecretKey,
+        pass_phrase: &[u8],
+    ) -> Result<EncryptedKeys, Error> {
+        let mut rng = thread_rng();
+        let mut eraser = Eraser::new();
+        let seed = &secret_key[..SEED_LENGTH];
+        eraser.add_suite::<Sodium>();
+        let pwbox = Sodium::build_box(&mut rng)
+            .seal(pass_phrase, &seed)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't create a pw box"))?;
```

Comment:
It's possible to use

```rust
let pwbox = Sodium::build_box(&mut rng)
    .seal(pass_phrase, &seed)
    .map_err(|e| Error::new(ErrorKind::Other, e.compat()))?;
```

here and elsewhere if you add `failure` to crate dependencies and import `failure::Fail` into the scope. Do you feel this would be an overkill?